components/merkledb/src/backends/temporarydb.rs

```diff
@@ -42,6 +42,7 @@ struct TemporarySnapshot {
 
 impl TemporaryDB {
     /// Creates a new, empty database.
+    #[must_use]
```

Comment:
🙃

Reply:
What about to disable this lint ?

Reply:
I think that this lint is useful in this component because it can warn about stupid mistakes.

Reply:
I may be an old-fashioned guy, but I don't see utility of `#[must_use]` everywhere; I'd suggest to use it only when the non-use of the return value is non-obvious to a developer *and* can result in logical errors (cf. `must_use` for `Result`s and `Future` / `Stream` implementations). From [the relevant lint description](https://rust-lang.github.io/rust-clippy/master/index.html#must_use_candidate):

> Why is this bad
>
> Not bad at all, this lint just shows places where you could add the attribute.